### PR TITLE
Parcel export format and discord jump for complex bounties

### PR DESCRIPTION
**Pull request by tescher**
Implements Parcel and full JSON formats for the export. The "Export" button now is a drop-down that asks which format.

The wallet address is stored in the User collection, so I used the mongoose populate functionality to do a join of the user data into the bounty data into a virtual field "payeeData". If we ever store the wallet address directly in the bounty, this will probably change, or go away altogether.

The other option was to create a User api, but having that public made me squeamish, so I made it at least a bit more obscure by doing the join on the backend.

I also modified the "Claim It" button so that if the bounty is multi-claimant or require-application, it just attempts to jump you to Discord to do the action since we don't support those on the web yet.

### CODE REVIEW

File: packages/react-app/src/components/pages/Bounties/index.tsx [hunk 1]
```diff
@@ -93,12 +94,22 @@ const SelectExport = ({
...
+					.filter(bounty => bounty.claimedBy)
+					.map(miscUtils.csvEncode);
+			} else {
+				csvData = bounties
+					.filter(({ _id }) => selectedBounties.includes(_id))
+					.map(miscUtils.csvEncode);
+			}
+			if (extraMapFunc) {
+				csvData = csvData.map(bounty => extraMapFunc(bounty));
+			}
```
Comment (jordaniza) [inline packages/react-app/src/components/pages/Bounties/index.tsx:112]: That's cool

File: packages/react-app/src/components/pages/Bounties/index.tsx [hunk 2]
```diff
@@ -130,6 +141,34 @@ const SelectExport = ({
 		}
 	};
 
+	const handleJSON = (): void => {
+
+		if (bounties) {
+
+		  // create file in browser
+		  const fileName = 'bounties.json';
```
Comment (jordaniza) [inline packages/react-app/src/components/pages/Bounties/index.tsx:149]: Might want to do `bounties-${new Date().now()}.json`

File: packages/react-app/src/components/pages/Bounties/index.tsx [hunk 3]
```diff
@@ -130,6 +141,34 @@ const SelectExport = ({
...
+		  const link = document.createElement('a');
+		  link.href = href;
+		  link.download = fileName + '.json';
+		  document.body.appendChild(link);
+		  link.click();
+		
+		  // clean up "a" element & remove ObjectURL
+		  document.body.removeChild(link);
+		  URL.revokeObjectURL(href);
+		}
```
Comment (jordaniza) [inline packages/react-app/src/components/pages/Bounties/index.tsx:168]: nice

File: packages/react-app/src/services/bounty.service.ts [hunk 4]
```diff
@@ -305,7 +306,19 @@ export const getBounties = async (
 		...paginationOptions,
 	};
 
-	return await Bounty.aggregateFn(bountyQuery);
+	const aggregation = await Bounty.aggregateFn(bountyQuery);
+
+	// Populate the payee data from the User DB. Needed until wallet addresses are stored directly in bounties
+	// Had to use a for loop so we didn't end up with a bunch of parallel db connections
```
Comment (jordaniza) [inline packages/react-app/src/services/bounty.service.ts:312]: Seems like you could await each operation before launching a new one?

**Review by jordaniza — APPROVED** [review]
Don't see any major issues with this - looks very sensible

File: packages/react-app/src/components/pages/Bounties/Bounty/claim.tsx [hunk 5]
```diff
@@ -153,26 +153,29 @@ const BountyClaim = ({ bounty }: { bounty: BountyCollection }): JSX.Element => {
...
-				Claim It
-			</Button>
+			<Box p={2}>
+				<Button
+					boxShadow={'md'}
+					transition="background 100ms linear"
+					bg={colorMode === 'light' ? 'primary.300' : 'primary.700'}
+					size='md'
+					width='200px'
+					disabled={!canonicalCard || !canonicalCard.channelId || !canonicalCard.messageId}
```
[the remark this answers is not in the document]
Comment (tescher) [reply packages/react-app/src/components/pages/Bounties/Bounty/claim.tsx:174]: Added to issues list

File: packages/react-app/src/services/bounty.service.ts (hunk 4 above)
Comment (tescher) [reply packages/react-app/src/services/bounty.service.ts:312]: I found many instances of people trying that, and it doesn't seem to work. This handles it OK

**Pull request merged**